iroha/src/permissions.rs

```diff
@@ -158,8 +159,43 @@ impl PermissionsValidator for CombinedPermissionsValidator {
     }
 }
 
-impl From<CombinedPermissionsValidator> for PermissionsValidatorBox {
-    fn from(validator: CombinedPermissionsValidator) -> Self {
+impl From<AllShouldSucceed> for PermissionsValidatorBox {
+    fn from(validator: AllShouldSucceed) -> Self {
+        Box::new(validator)
+    }
+}
+
+/// A container for multiple permissions validators. It will succeed if any validator succeeds.
+#[allow(missing_debug_implementations)]
+pub struct AnyShouldSucceed {
+    name: String,
+    validators: Vec<PermissionsValidatorBox>,
+}
+
+impl PermissionsValidator for AnyShouldSucceed {
+    fn check_instruction(
+        &self,
+        authority: AccountId,
+        instruction: Instruction,
+        wsv: &WorldStateView,
+    ) -> Result<(), DenialReason> {
+        for validator in &self.validators {
+            if validator
+                .check_instruction(authority.clone(), instruction.clone(), wsv)
```

Comment:
Should they passed here by reference? To avoid cloning too much?

Reply:
I will check

Reply:
I am planning to check it and correct if necessary in a separate PR if you don't mind. It is a bit more general change than what we have here as it would restructure all Permission Validators.